src/_reduceLazy.ts

```diff
@@ -25,14 +25,17 @@ export function _reduceLazy<T, K>(
   array: Array<T>,
   lazy: (item: T, index?: number, array?: Array<T>) => LazyResult<K>,
   indexed?: boolean
-) {
-  return array.reduce((acc: Array<K>, item, index) => {
+): Array<K> {
+  const newArray: Array<K> = [];
+  // We intentionally use a for loop here instead of reduce for performance reasons. See https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/ for more info
+  for (let index = 0; index < array.length; index++) {
```

Comment:
You should use a `for...of` loop (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) here. It's stricter as it doesn't use an index into the array (see #291) and is transpiled to the same code on es5 anyway.

Reply:
Hey, sorry for the mistaken suggestions, you can ignore them, while we compile down to es5 they aren't relevant. Your solution is on-point in the current configuration.